clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5087,6 +5087,27 @@ static void handleSYCLRegisterNumAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   D->addAttr(::new (S.Context) SYCLRegisterNumAttr(S.Context, AL, RegNo));
 }
 
+static void handleSYCLIntelESimdWidenAttrAttr(Sema &S, Decl *D,
+                                              const ParsedAttr &Attr) {
+  uint32_t NumSimdLanes = 0;
+  const Expr *E = Attr.getArgAsExpr(0);
+  if (!checkUInt32Argument(S, Attr, E, NumSimdLanes, 0,
+                           /*StrictlyUnsigned=*/true))
+    return;
+
+  if (NumSimdLanes != 8 && NumSimdLanes != 16 && NumSimdLanes != 32) {
+    S.Diag(Attr.getLoc(), diag::err_sycl_esimd_widen_unsupported_value)
+        << Attr << E->getSourceRange();
+    return;
+  }
+
+  if (D->getAttr<SYCLIntelESimdWidenAttr>())
```

Comment:
check D->hasAttr instead?

Reply:
FWIW, this won't catch the issue on redeclarations (see the comment above from @elizabethandrews).

Reply:
Reimplemented.